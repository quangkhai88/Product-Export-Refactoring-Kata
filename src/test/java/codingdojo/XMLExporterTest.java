package codingdojo;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.List;

import static codingdojo.SampleModelObjects.*;

class XMLExporterTest {

    @Test
    void testFullExport() {
        List<Order> list = Arrays.asList(RecentOrder, OldOrder);
        String xml = XMLExporter.exportFull(list);
        Approvals.verifyXml(xml);
    }

    @Test
    void testTaxExport() {
        List<Order> list = Arrays.asList(RecentOrder, OldOrder);
        String xml = XMLExporter.exportTaxDetails(list);
        Approvals.verifyXml(xml);
    }

    @Test
    void testFullExportWithEvent() {
        List<Order> list = Arrays.asList(RecentOrder, OldOrder);
        RecentOrder.addProducts(Masterclass);
        String xml = XMLExporter.exportFull(list);
        Approvals.verifyXml(xml);
    }

    @Test
    void testFullExportWithEventAndWeight() {
        List<Order> list = Arrays.asList(RecentOrder, OldOrder);
        RecentOrder.addProducts(Masterclass);
        String xml = XMLExporter.exportFull(list);
        Approvals.verifyXml(xml);
    }


    @Test
    void testExportStore() {
        List<Order> list = Arrays.asList(RecentOrder, OldOrder);
        String xml = XMLExporter.exportFull(list);
        Approvals.verifyXml(xml);
    }

    @Test
    void testExportHistory() {
        List<Order> list = Arrays.asList(RecentOrder, OldOrder);
        String xml = XMLExporter.exportFull(list);
        Approvals.verifyXml(xml);
    }
}
