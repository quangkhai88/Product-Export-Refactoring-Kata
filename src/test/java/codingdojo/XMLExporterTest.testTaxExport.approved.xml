<?xml version="1.0" encoding="UTF-8"?><orderTax>
  <order date="2018-09-01T00:00Z">
    <product id="EVENT02">Makeover</product>
    <product id="EVENT01">Eyeshadow Masterclass</product>
    <orderTax currency="USD">87.50</orderTax>
  </order>
  <order date="2017-09-01T00:00Z">
    <product id="LIPSTICK01">Cherry Bloom</product>
    <orderTax currency="USD">12.62</orderTax>
  </order>
  100.12
</orderTax>
